mllib/src/main/scala/org/apache/spark/ml/feature/Bucketizer.scala

```diff
@@ -105,20 +106,21 @@ final class Bucketizer @Since("1.4.0") (@Since("1.4.0") override val uid: String
     transformSchema(dataset.schema)
     val (filteredDataset, keepInvalid) = {
       if (getHandleInvalid == Bucketizer.SKIP_INVALID) {
-        // "skip" NaN option is set, will filter out NaN values in the dataset
+        // "skip" NaN/NULL option is set, will filter out NaN/NULL values in the dataset
         (dataset.na.drop().toDF(), false)
       } else {
         (dataset.toDF(), getHandleInvalid == Bucketizer.KEEP_INVALID)
       }
     }
 
-    val bucketizer: UserDefinedFunction = udf { (feature: Double) =>
+    val bucketizer: UserDefinedFunction = udf { (row: Row) =>
```

Comment:
I believe you should try to avoid using a udf on a row because the serialization costs will be more expensive... hmm how could we make this perform well and handle nulls?  Does it work with Option[Double] instead of Row?